Fix Kubernetes Secret Encoding Under Python 3

morinap opened the pull request:
### What does this PR do?
Base64 encode secret data properly under Python3. Secret data sent to the Kubernetes API must be a base64 encoded String object.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/55292

### Previous Behavior
A fatal error occurred

### New Behavior
Resolve fatal error

### Tests written?
No

### Commits signed with GPG?
Yes

dwoz commented on salt/modules/kubernetesmod.py:
```diff
@@ -1593,3 +1593,13 @@ def __enforce_only_strings_dict(dictionary):
+    Converts a string of data into a base64 encoded string based on python version
+    '''
+    if sys.version_info >= (3, 0):
+        return str(base64.b64encode(string.encode('utf-8')), 'utf-8')
```
How do we know this will always be `utf-8`?

morinap replied:
@dwoz If `string` in this case is already a `str` object and I'm simply trying to get it to a bytes representation so that can be passed to the `b64encode` call, does it matter what the underlying encoding of the string is? Perhaps I am misunderstanding the problem. Are you suggesting there may be a problem if there are characters in `string` that cannot be encoded by `utf-8`?

DmitryKuzmenko replied:
@morinap There are 2 more salty ways to do this depending on your goals. If you need a `unicode` string as a result on both py2 and py3 you can use
```python
salt.utils.hashutils.base64_b64encode(string)
```
If you need a native `str` object on all platforms you can do either
```python
salt.utils.stringutils.to_str(salt.utils.hashutils.base64_b64encode(string))
```

morinap committed:
Add base unit test for create and replace secret

morinap committed:
Base64 encode secret data properly under Python3

Secret data sent to the Kubernetes API must be a base64 encoded String object

resolves #55292

morinap commented:
@dwoz I've added the appropriate unit test cases to this PR, rebased, and pushed everything up. I'll await a response from you regarding the encoding question.

DmitryKuzmenko commented:
@morinap thank you for contribution. Special thanks for the test!
Now this needs the following steps to be applied:
1. Update your master branch and install pre-commit hook as described in the contribution guide https://docs.saltstack.com/en/latest/topics/development/contributing.html
2. Rebase your commit resolving conflicts and update your fix with my comment suggestion.
3. Run pre-commit run on your branch to re-format your code (quotes and spacing) it will do it automatically.
